plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/mapping/RemoteSchemaNamesProvider.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.jdbc.mapping;
+
+import java.util.Collection;
+
+@FunctionalInterface
+public interface RemoteSchemaNamesProvider
```

Comment:
I would merge all these interfaces along with DatabaseMetaDataRemoteIdentifierSupplier

Reply:
But how it will help?
My main problem is that we have sql.connection in interface - IdentifierMapping.
If I remove it, then I need to pass lambda (like ability to fetch schemas/tables).

Reply:
I think it will help because you can define an entity per connector. For Druid you do this, for mongo this, for postegres this. 

Lambda is nice if you often change the implementation of it. Here it is domain specific thing related to remote data source.

It is just a comment. Up to you for now let's see how it goes with your approach.

Reply:
I also don't like lambda here, but because we tightly coupled with sql.connection in interface, and we need ability to lazily fetch Schemas/Tables I don't see more cleaner way :(

Reply:
See https://github.com/trinodb/trino/pull/17996#discussion_r1244237044